rak7391/rak7391.go

```diff
@@ -0,0 +1,996 @@
+		r.region = regions.EU
+	}
+
+	b, err := board.FromDependencies(deps, cfg.BoardName)
+	if err != nil {
+		return err
+	}
+
+	if cfg.Concentrator1 != nil {
+		if err := r.createConcentrator(ctx, "rak1", rak7391ResetPin1, b, cfg.Concentrator1); err != nil {
+			return err
+		}
+		r.logger.Debugf("created concentrator 1")
+	}
+
+	if cfg.Concentrator2 != nil {
+		if err := r.createConcentrator(ctx, "rak2", rak7391ResetPin2, b, cfg.Concentrator2); err != nil {
+			return err
+		}
+		r.logger.Debugf("created concentrator 2")
+	}
+	r.workers = utils.NewBackgroundStoppableWorkers(r.receivePackets)
+	return nil
+}
+
+func (r *rak7391) createConcentrator(ctx context.Context,
+	id, restPinName string,
+	b board.Board,
+	cfg *ConcentratorConfig,
+) error {
+	rstPin, err := b.GPIOPinByName(restPinName)
+	if err != nil {
+		return err
+	}
+	c := concentrator{rstPin: rstPin}
+
+	// defaults if nothing configured
+	path := "/dev/spidev0.0"
+	comType := spi
+
+	if cfg.Bus != nil {
+		path = fmt.Sprintf("/dev/spidev0.%d", *cfg.Bus)
+		comType = spi
+	}
+
+	if cfg.Path != "" {
+		// Resolve the symlink
+		if strings.Contains(cfg.Path, "by-path") || strings.Contains(cfg.Path, "by-id") {
+			resolvedPath, err := filepath.EvalSymlinks(cfg.Path)
+			if err != nil {
+				return fmt.Errorf("failed to resolve symlink of path %s: %w", cfg.Path, err)
+			}
+			cfg.Path = resolvedPath
+		}
+		path = cfg.Path
+		comType = usb
+	}
+
+	err = resetGateway(ctx, rstPin, nil)
+	if err != nil {
+		return fmt.Errorf("error initializing the gateway: %w", err)
+	}
+
+	args := []string{
+		fmt.Sprintf("--comType=%d", comType),
+		"--path=" + path,
+		fmt.Sprintf("--region=%d", r.region),
+	}
+
+	var cgoexe string
+	cgoexe = r.cgoPath
+	if r.cgoPath == "" {
+		dir, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("error getting current directory: %w", err)
+		}
+
+		cgoexe = dir + "/cgo"
+		_, err = os.Stat(cgoexe)
+		if err != nil {
+			return fmt.Errorf("error getting the cgo exe: %w", err)
+		}
+	}
+
+	logReader, logWriter := io.Pipe()
+	bufferedLogReader := bufio.NewReader(logReader)
+	c.pipeWriter = logWriter
+	c.pipeReader = logReader
+	c.logReader = bufferedLogReader
+
+	config := pexec.ProcessConfig{
+		ID:        id,
+		Name:      cgoexe,
+		Args:      args,
+		Log:       false,
+		LogWriter: logWriter,
+	}
+
+	process := pexec.NewManagedProcess(config, r.logger)
+	c.process = process
+
+	if err := process.Start(ctx); err != nil {
+		return fmt.Errorf("failed to start binary: %w", err)
+	}
+
+	port, err := c.watchLogs(ctx, r.logger)
+	if err != nil {
+		return err
+	}
+
+	r.logger.Debugf("connecting to port: %s\n", port)
+	target := "localhost:" + port
+	conn, err := grpc.NewClient(target, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return fmt.Errorf("error connecting to server: %w", err)
+	}
+
+	r.logger.Info("connected to client")
+
+	c.conn = conn
+	c.client = pb.NewSensorServiceClient(conn)
+	c.started = true
```

Comment:
When is this ever set to false in this driver's code? Just on creation? So viam server restart?